src/vid2805-dual-stepper-element.ts

```diff
@@ -0,0 +1,191 @@
+import { html, LitElement, svg } from 'lit';
+import { customElement, property } from 'lit/decorators.js';
+import { ElementPin } from '.';
+import { mmToPix } from './utils/units';
+
+export type HandShape = 'arrow' | 'plain'; // | 'fancy';
+
+export class StepperHand {
+  /**
+   * length used to control hand length
+   */
+  length = 25;
+
+  /**
+   * colour used to paint the hand
+   */
+  colour = '';
+
+  /**
+   * hand shape selects from predefined shape paths
+   */
+  shape: HandShape = 'plain';
+
+  /**
+   * the current hand angle
+   */
+  angle = 0;
+
+  constructor(length = 25, colour = '', shape: HandShape = 'plain', angle = 0) {
+    this.angle = angle;
+    this.colour = colour;
+    this.shape = shape;
+    this.length = length;
+  }
+}
+
+@customElement('wokwi-vid2805-dual-stepper')
+export class VID2805DualStepperElement extends LitElement {
+  @property() outerHand = new StepperHand();
+  @property() innerHand = new StepperHand();
+
+  private coords() {
+    const _x = 112.3; // x location of shaft point
+    const _y = 50; // y location of shaft point
+    const _innerOff = 4.7; // offset to center of inner hand's ring
+    const _outerOff = 9; // offset to center of outer hand's ring
+    let _trY = 0;
+
+    const ml = Math.max(this.innerHand.length, this.outerHand.length);
+    if (ml > 30) {
+      _trY = ml - 30;
+    }
+    return {
+      x: _x,
+      y: _y,
+      innerOff: _innerOff,
+      outerOff: _outerOff,
+      trY: _trY,
+    };
+  }
+
+  get pinInfo(): ElementPin[] {
+    const { x, y, innerOff, outerOff, trY } = this.coords();
+    const pinXY = (x: number) => {
+      return { x: 15 + trY, y: (48.5 - x * 2.54) * mmToPix };
+    };
+
+    const pi = [
+      { name: 'A1+', ...pinXY(0), number: 1, signals: [] },
+      { name: 'A1-', ...pinXY(1), number: 2, signals: [] },
+      { name: 'B1+', ...pinXY(2), number: 3, signals: [] },
+      { name: 'B1-', ...pinXY(3), number: 4, signals: [] },
+      { name: 'A2+', ...pinXY(4), number: 5, signals: [] },
+      { name: 'A2-', ...pinXY(5), number: 6, signals: [] },
+      { name: 'B2+', ...pinXY(6), number: 7, signals: [] },
+      { name: 'B2-', ...pinXY(7), number: 8, signals: [] },
+    ];
+
+    console.debug(pi);
+    return pi;
+  }
+
+  readonly handMap: { [key: string]: string } = {
+    outer_plain_hand:
```

Comment:
key names should be camelCase, e.g. `outerPlainHand`.

Reply:
done